spinalcordtoolbox/utils.py

```diff
@@ -138,83 +156,19 @@ def is_exe(fpath):
     return None
 
 
-def __get_branch():
+def extract_fname(fpath):
     """
-    Fallback if for some reason the value vas no set by sct_launcher
-    :return:
-    """
-
-    p = subprocess.Popen(["git", "rev-parse", "--abbrev-ref", "HEAD"], stdout=subprocess.PIPE,
-                         stderr=subprocess.PIPE, cwd=__sct_dir__)
-    output, _ = p.communicate()
-    status = p.returncode
-
-    if status == 0:
-        return output.decode().strip()
-
+    Split a full path into a parent folder component, filename stem and extension.
 
-def __get_commit():
-    """
-    :return: git commit ID, with trailing '*' if modified
+    Note: for .nii.gz the extension is understandably .nii.gz, not .gz
+    (``os.path.splitext()`` would want to do the latter, hence the special case).
     """
-    p = subprocess.Popen(["git", "rev-parse", "HEAD"], stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                         cwd=__sct_dir__)
-    output, _ = p.communicate()
-    status = p.returncode
-    if status == 0:
-        commit = output.decode().strip()
-    else:
-        commit = "?!?"
-
-    p = subprocess.Popen(["git", "status", "--porcelain"], stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                         cwd=__sct_dir__)
-    output, _ = p.communicate()
-    status = p.returncode
-    if status == 0:
-        unclean = True
-        for line in output.decode().strip().splitlines():
-            line = line.rstrip()
-            if line.startswith("??"): # ignore ignored files, they can't hurt
-               continue
-            break
-        else:
-            unclean = False
-        if unclean:
-            commit += "*"
-
-    return commit
-
-
-def _git_info(commit_env='SCT_COMMIT', branch_env='SCT_BRANCH'):
-
-    sct_commit = os.getenv(commit_env, "unknown")
-    sct_branch = os.getenv(branch_env, "unknown")
-    if check_exe("git") and os.path.isdir(os.path.join(__sct_dir__, ".git")):
-        sct_commit = __get_commit() or sct_commit
-        sct_branch = __get_branch() or sct_branch
-
-    if sct_commit is not 'unknown':
-        install_type = 'git'
+    parent, filename = os.path.split(fpath)
+    if filename.endswith(".nii.gz"):
+        stem, ext = filename[:-7], ".nii.gz"
```

Comment:
You could add the same special-case for `.tar.gz`. The number of special cases in here might fast get unwieldy.

Reply:
It almost looks like this exists in the stdlib @ https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffixes but it only does half the job. :abacus:

Reply:
What about

```suggestion
    dir, filename = os.path.split(filename)
    stem, ext = filename.split(".",1)
    ext = "." + ext
    return os.path.join(dir, stem), ext
```

Reply:
the problem with that implementation is that it fails the following test:
```python
    assert utils.splitext('nice.image.nii.gz') == ('nice.image', '.nii.gz')
```
I've seen such file names in my life 🙄...

Reply:
now accommodating tar.gz (and we can add to the list of extensions if we encounter more): f537788664a3fb81876235d67f92ef910f1bd6f6

Reply:
> the problem with that implementation is that it fails the following test:
> 
> ```python
>     assert utils.splitext('nice.image.nii.gz') == ('nice.image', '.nii.gz')
> ```
> 
> I've seen such file names in my life roll_eyes...

I was hoping they wouldn't be an issue. Alright, special-case away!